src/formatting_layer.rs

```diff
@@ -313,9 +315,14 @@ where
     W: for<'a> MakeWriter<'a> + 'static,
 {
     fn on_event(&self, event: &Event<'_>, ctx: Context<'_, S>) {
-        // Events do not necessarily happen in the context of a span, hence lookup_current
-        // returns an `Option<SpanRef<_>>` instead of a `SpanRef<_>`.
-        let current_span = ctx.lookup_current();
+        // Lookup the current span, unless the event is a "root" event.
+        let current_span = if event.is_root() {
+            None
+        } else {
+            // Events do not necessarily happen in the context of a span, hence lookup_current
+            // returns an `Option<SpanRef<_>>` instead of a `SpanRef<_>`.
+            ctx.lookup_current()
+        };
```

Comment:
I'm not sure this is correct.
I think we should:

- Look up the current span if https://docs.rs/tracing/latest/tracing/event/struct.Event.html#method.is_contextual returns true
- Look up the parent if https://docs.rs/tracing/latest/tracing/event/struct.Event.html#method.parent returns `Some`
- Set it to `None` if all the above didn't work

Reply:
Right, I think that makes sense.